src/ConnectionAutomaton.cpp

```diff
@@ -60,8 +60,7 @@ void ConnectionAutomaton::connect(
     std::shared_ptr<FrameTransport> frameTransport,
     bool sendingPendingFrames) {
   CHECK(isDisconnectedOrClosed());
-  CHECK(frameTransport);
-  CHECK(!frameTransport->isClosed());
+  CHECK(frameTransport && !frameTransport->isClosed());
```

Comment:
I don't think overhead is a real concern here, and the amount of code should be infinitesimal. Since these are the kinds of lines that show up in crash logs, I really think it's important to have as precise a root cause as possible—a combined binary condition like this one means that we'll have two possible causes instead of one, and so makes it harder to debug.